packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -28,27 +28,38 @@ export function findOffsetInDOM(
 ): FindOffsetResults {
   const allNodes = Array.from(parentNode.querySelectorAll('[data-child-begin]')) as HTMLElement[];
 
-  let beginNode,
-    endNode,
-    idx = 0;
-
-  do {
-    beginNode = allNodes[idx];
-  } while (
-    ++idx &&
-    idx < allNodes.length &&
-    begin >= parseInt(allNodes[idx].dataset.childBegin || '0', 10)
-  );
+  let beginNode = null,
```

Comment:
Can you provide more details about why this was changed?

Reply:
I ended up doing this differently, as even what I discussed above had at least one corner case that it would have missed. The new approach is more of a tree traversal with the help of a binary search to tell us where to "drill down" to look for each descendant. With the optimizations, this procedure tends to take <1ms to complete (if I'm not mistaken, it used to take on the order of ~5ms). Let me know if it's not clear from the comments how that procedure works.